### PR TITLE
Added Framework and Webkit related test cases

### PR DESCRIPTION
**Pull request by sandeep-vedam**
Converted test cases related to Framework and Webkit module in Metromatic branch as per the latest framework

### CODE REVIEW

File: src/FrameWorkController_002.test.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  getControllerPluginData,
+  pluginActivate,
+  pluginDeactivate,
+  restartFramework,
+} from './commonMethods/commonFunctions'
+
+export default {
+  title: 'Framework Controller test 02',
+  description: 'Tests basic functionality of the controller module',
+  teardown: restartFramework,
```
Comment (wouterlucas) [inline src/FrameWorkController_002.test.js:11]: I'd suggest to remove this. This was an artifact of a crashing WPEFramework, no need to do this anymore. WPEFramework should continue to function and not need a reboot after this test.

File: src/Framework_DeviceInfo_001.test.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  pluginDeactivate,
+  pluginActivate,
+  restartFramework,
+} from './commonMethods/commonFunctions'
+import constants from './commonMethods/constants'
+
+export default {
+  title: 'Framework Device Info test',
+  description: 'Validates functionality of the Device Info plugin',
+  teardown: restartFramework,
```
Comment (wouterlucas) [inline src/Framework_DeviceInfo_001.test.js:12]: Same here

File: src/Framework_Monitor_001.test.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  pluginDeactivate,
+  pluginActivate,
+  restartFramework,
+} from './commonMethods/commonFunctions'
+import constants from './commonMethods/constants'
+
+export default {
+  title: 'Framework Monitor test 001',
+  description: 'Tests if the Framework Monitor module works',
+  teardown: restartFramework,
```
Comment (wouterlucas) [inline src/Framework_Monitor_001.test.js:12]: Same here

File: src/Framework_OpenCDM_001.test.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  restartFramework,
+} from './commonMethods/commonFunctions'
+import constants from './commonMethods/constants'
+
+export default {
+  title: 'OCDM startup robustness test',
+  description:
+    'Starts and stops the OCDM plugin repeatedly and checks if everything is started correctly',
+  repeat: 30,
+  teardown: restartFramework,
```
Comment (wouterlucas) [inline src/Framework_OpenCDM_001.test.js:15]: Same here

File: src/Framework_Provisioning_001.test.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  pluginDeactivate,
+  pluginActivate,
+  restartFramework,
+} from './commonMethods/commonFunctions'
+import constants from './commonMethods/constants'
+
+export default {
+  title: 'Framework provision test 001',
+  description: 'Tests if the Framework provision module works',
+  teardown: restartFramework,
```
Comment (wouterlucas) [inline src/Framework_Provisioning_001.test.js:12]: Same here

File: src/Framework_Provisioning_002.test.js [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+  pluginActivate,
+  restartFramework,
+  putRequestForPlugin,
+} from './commonMethods/commonFunctions'
+import constants from './commonMethods/constants'
+
+export default {
+  title: 'Framework provision test 002',
+  description: 'Check if device is provisioned',
+  teardown: restartFramework,
```
Comment (wouterlucas) [inline src/Framework_Provisioning_002.test.js:13]: Same here

File: src/Framework_Snapshot_004.test.js [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+      () => webKitBrowserOps.call(this),
+      () =>
+        (listener = this.$thunder.api.WebKitBrowser.on('urlchange', data => {
+          this.$data.write('currentUrl', data.url)
+        })),
+    ])
+  },
+  teardown() {
+    listener.dispose()
+    restartFramework.call(this)
```
Comment (wouterlucas) [inline src/Framework_Snapshot_004.test.js:31]: Same here

File: src/commonMethods/commonFunctions.js [hunk 8]
```diff
@@ -0,0 +1,296 @@
+import constants from './constants'
```
Comment (wouterlucas) [inline src/commonMethods/commonFunctions.js:1]: The commonFunctions should be moved to Storm CORE

File: src/commonMethods/commonFunctions.js [hunk 9]
```diff
@@ -0,0 +1,296 @@
...
+  ])
+}
+
+/**
+ * This function performs below operations on WebKitBrowser Plugin
+ *  - Deactivate
+ *  - Activate
+ *  - Resume
+ */
+export const webKitBrowserOps = function() {
```
Comment (wouterlucas) [inline src/commonMethods/commonFunctions.js:246]: Suggest to rename this for a more readable name, e.g. webKitBrowserRestart or RestartAndResume

**Review by wouterlucas — CHANGES_REQUESTED** [review]
Some minor feedback, otherwise looks really good!

File: src/commonMethods/commonFunctions.js (hunk 9 above)
Comment (sandeep-vedam) [reply src/commonMethods/commonFunctions.js:246]: Renamed function name to a readable name as 'webKitBrowserStartAndResume'

File: src/Framework_Snapshot_004.test.js (hunk 7 above)
Comment (sandeep-vedam) [reply src/Framework_Snapshot_004.test.js:31]: Removed restart Framework

**Review by sandeep-vedam — COMMENTED** [review]
This PR is splitted into multiple PR's. This can be ignored

**Pull request merged**